doc/user/content/concepts/indexes.md

```diff
@@ -17,52 +17,56 @@ In Materialize, indexes represent query results stored in memory within a
-computations. For queries that have no supporting indexes, Materialize uses the
-same mechanics used by indexes to optimize computations.
+In Materialize, you can create indexes on sources to maintain in-memory
+up-to-date source data.
 
-Specific instances where indexes can be useful to improve performance include:
+```mzsql
+CREATE INDEX idx_on_my_source ON my_source (...);
+```
 
-- When used in ad-hoc queries.
+## Indexes on views
 
-- When used by multiple queries within the same cluster.
+In Materialize, indexes on views both **compute, and as new data arrives,
+incrementally update view results in memory** within that
+[cluster](/concepts/clusters/). Within the cluster, the in-memory up-to-date
+results are immediately available and computationally free to query.
 
-- When used to enable delta joins.
+In Materialize, indexes on a [view](/concepts/views/#views) **compute and, as
+new data arrives, incrementally update** view results in memory within a
+[cluster](/concepts/clusters/) instead of recomputing the results from scratch.
 
-For more information, see [Optimization](/transform-data/optimization).
+Within the cluster, the in-memory up-to-date results are immediately available
+and computationally free to query.
 
-{{</ tip >}}
+```mzsql
+CREATE INDEX idx_on_my_view ON my_view_name(...) ;
+```
 
-## Indexes and views
+For considerations on using indexes as well as using indexes vs. materialized
+views, see [Usage patterns](#usage-patterns).
 
-In Materialize, indexes on a [view](/concepts/views/#views) **maintain and
-incrementally update** view results in memory within a
-[cluster](/concepts/clusters/). The in-memory up-to-date results are accessible
-to queries within the cluster.
+## Indexes on materialized views
 
-## Indexes and materialized views
+In Materialize, materialized views already maintain the up-to-date results in
+durable storage. Indexing a [materialized
+view](/concepts/views/#materialized-views) loads the already up-to-date results
+into memory; that is, the indexes on materialized views are able to serve
+up-to-date results without themselves performing the incremental updates.
+However, unlike materialized views, indexes are accessible only within a
+[cluster](/concepts/clusters/).
 
-In Materialize, indexing a
-[materialized view](/concepts/views/#materialized-views) maintains results in memory within the [cluster](/concepts/clusters/). Because
-materialized views maintain the up-to-date results in durable storage, indexes
-on materialized views serve up-to-date results without themselves performing the
-incremental computation. However, unlike the materialized view, indexes are
-local to the cluster.
+For considerations on using indexes as well as using indexes vs. materialized
+views, see [Usage patterns](#usage-patterns).
 
 ```mzsql
-CREATE INDEX idx_on_my_view ON my_view_name(...) ;
+CREATE INDEX idx_on_my_mat_view ON my_mat_view_name(...) ;
```

Comment:
I quite like it.  Not sure if I would put it here in the concepts page though.  It seems like a nice hands-on exercise during training as well as some other page (although, since currently lacking the training and the "some other page" ... I'll see how best to incorporate this as I go through other feedback comments.)